### PR TITLE
feat: add member to context

### PR DESCRIPTION
**Pull request by 0c370t**

### CODE REVIEW

File: src/marshal/decorators/command.decorator.ts [hunk 1]
```diff
@@ -1,18 +1,41 @@
...
+            },
+            relations: ["profile"],
+        });
+        const members = await lastValueFrom(memberRx) as unknown as TempMember[];
+        const context = {...params[1]};
+        if (members.length) {
+            context.member = members[0];
+            if (members.length > 1) {
+                logger.error("More than one members with given discord id found!");
+            }
```
Comment (zachstence) [inline src/marshal/decorators/command.decorator.ts:36]: Just a semantic preference, but move the `if` error block before the `context.member` assignment

File: src/marshal/marshal.spec.ts [hunk 2]
```diff
@@ -59,7 +59,7 @@ describe("Marshal", () => {
 
         it("should correctly register it's command when instantiated", () => {
             expect(managerService.commands.size).toBe(1);
-            managerService.commands.get("mycommand")?.func(undefined as unknown as Message);
+            managerService.commands.get("mycommand")?.func(undefined as unknown as Message, {});
             expect(commandHandlerSpy).toBeCalled();
         });
```
Comment (zachstence) [inline src/marshal/marshal.spec.ts:65]: We need test(s) asserting that the context/member are correctly given to the CommandFunction

File: src/marshal/types.ts [hunk 3]
```diff
@@ -5,11 +5,16 @@ export enum MarshalMetadataKey {
     Command = "marshal:commandMetas",
     CommandNotFound = "marshal:commandNotFoundMetas",
 }
+/**
+ * This type is temporary, and will be replaced when @sprocketbot/entities becomes available
```
Comment (zachstence) [inline src/marshal/types.ts:9]: Put a `TODO` here so we don't lose this

File: src/marshal/types.ts [hunk 4]
```diff
@@ -5,11 +5,16 @@ export enum MarshalMetadataKey {
...
+ */
+export interface TempMember {discordId: string; profile?: {name: string;};}
 export interface MarshalCommandContext {
     args?: Record<string, string>;
+    member?: TempMember;
```
Comment (zachstence) [inline src/marshal/types.ts:14]: Would `author` be a better name to indicate that this member on the context is the one who used the command?

**Review by zachstence — CHANGES_REQUESTED** [review]

File: src/marshal/types.ts (hunk 3 above)
Comment (0c370t) [reply src/marshal/types.ts:9]: @zachstence we probably still need to figure this out

**Pull request merged**